gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -113,6 +114,11 @@ trait BasicScanExecTransformer extends LeafTransformSupport with BaseDataSource
     }.asJava
     // Will put all filter expressions into an AND expression
     val transformer = filterExprs()
+      .map {
+        case ar: AttributeReference if ar.dataType == BooleanType =>
```

Comment:
does it work with not ? e.g., `WHERE !b`

Reply:
yes, `where !b` equal to `where b=null`